src/SFtp.cc

```diff
@@ -1191,8 +1194,6 @@ int SFtp::Read(Buffer *buf,int size)
       int bytes_allowed=rate_limit->BytesAllowedToGet();
       if(size1>bytes_allowed)
 	 size1=bytes_allowed;
-      if(size1==0)
```

Comment:
Ok, now I see that bytes_allowed might be zero, so it would make sense to keep that check in. I'll rebase the commit.